libkineto/src/RoctracerActivityApi.cpp

```diff
@@ -235,127 +235,72 @@ int RoctracerActivityApi::processActivities(
-        name = demangle(hipKernelNameRef(item.function));
-      }
-      if (!name.empty()) {
-        uint32_t string_id = reverseStrings_[name];
-        if (string_id == 0) {
-          string_id = nextStringId_++;
-          reverseStrings_[name] = string_id;
-          strings_[string_id] = name;
-        }
-        kernelNames_[item.id] = string_id;
-      }
-
       logger.handleGenericActivity(a);
       ++count;
     }
   }  // isLogged(ActivityType::CUDA_RUNTIME)
 
   // Async Ops
 
-  for (auto& buffer : *d->gpuTraceBuffers_) {
-    const roctracer_record_t* record = (const roctracer_record_t*)(buffer.data_);
-    const roctracer_record_t* end_record = (const roctracer_record_t*)(buffer.data_ + buffer.validSize_);
+  for (auto &item : d->opRows_) {
+    if (!inRange(startTime, endTime, item.begin))
+          continue;
     GenericTraceActivity a;
 
-    while (record < end_record) {
-      if (record->domain == ACTIVITY_DOMAIN_HIP_API) {
-        const char *name = roctracer_op_string(record->domain, record->op, record->kind);
-        a.device = record->process_id;
-        a.resource = record->thread_id;
-
-        a.startTime = (record->begin_ns + toffset) / 1000;
-        a.endTime = (record->end_ns + toffset) / 1000;
-        a.id = record->correlation_id;
-
-        a.activityType = ActivityType::CUDA_RUNTIME;
-        a.activityName = std::string(name);
-        a.flow.id = record->correlation_id;
-        a.flow.type = kLinkAsyncCpuGpu;
-        a.flow.start = true;
-
-        auto it = externalCorrelations.find(a.id);
-        a.linked = linkedActivity(it == externalCorrelations.end() ? 0 : it->second);
-
-        if (inRange(startTime, endTime, record->begin_ns)) {
-            logger.handleGenericActivity(a);
-            ++count;
-        }
-      }
-      else if (record->domain == ACTIVITY_DOMAIN_HCC_OPS) {
-        // Overlay launch metadata for kernels
-        auto kit = kernelLaunches_.find(record->correlation_id);
-        if (kit != kernelLaunches_.end()) {
-          a = (*kit).second;
-        }
-
-        const char *name = roctracer_op_string(record->domain, record->op, record->kind);
-        a.device = record->device_id;
-        a.resource = record->queue_id;
-
-        a.startTime = (record->begin_ns + toffset) / 1000;
-        a.endTime = (record->end_ns + toffset) / 1000;
-        a.id = record->correlation_id;
+    // Overlay launch metadata for kernels
+    auto kit = kernelLaunches_.find(item.id);
+    if (kit != kernelLaunches_.end()) {
+      a = (*kit).second;
+    }
 
+    const char *name = roctracer_op_string(item.domain, item.op, item.kind);
+    a.device = item.device;
+    a.resource = item.queue;
+
+    a.startTime = (item.begin + toffset) / 1000;
+    a.endTime = (item.end + toffset) / 1000;
+    a.id = item.id;
+
+    a.activityType = ActivityType::CONCURRENT_KERNEL;
+    a.activityName = item.kernelName.length() > 0 ? item.kernelName : std::string(name);
```

Comment:
@mwootton in that application, item.kernelName is empty and name is "Task"

Reply:
Yes, that is a different issue with looking up some kernel names.  The will come as a rocm fix.
The fact is says "Task" means I'm looking it up correctly, right? :)

Reply:
The original code seems to just get the name from roctracer_op_string and this change will first use item.kernelName, is that right? Why the item.kernelName is empty?